### ISSUE TITLE
Car driving out of the lane but output is success

### ISSUE THREAD

**Issue by ezecastellano** [se2p/tool-competition-av#43]
We have a question regarding the output of the tool. 

In the description of the competition, it says that "a test fail if the ego-car does not move, or does not reach the end of the road within a timeout (computed over the length of the road), or drives off the lane."

Given the path [(10, 100), (40, 113), (60, 115), (80, 99), (100, 80), (120, 70), (140, 58), (160, 38), (180, 30)], the ego car goes out of the lane in the very first curve. However, the test continues to the end and the output of the test is "success". Shouldn't the output of this test be failed? What does exactly mean to "drive off the lane"? We assume that crossing the yellow line should also be a failure. 

Also, does the test always continue to the end even if there is a failure at an early stage? Or should the test actually stop around 00:08s, when the car goes out of the lane?

https://user-images.githubusercontent.com/5766144/102904985-db109b00-44b5-11eb-93b8-0d8a2b62d388.mov

Thank you.

**Comment by alessiogambi** [comment]
@ezecastellano thanks for the video and the clear explanation. 

**TL;DR**
the test does not fail because the car does not go enough outside the lane.

**Long Version**
Let me try to answer your questions:

> What does exactly mean to "drive off the lane"? 

This is trickier than it looks. A very stringent formulation of "driving off the lane" would be simply that the car crosses any line (like in your video); however, in our experience, this definition might be too stringent and might report that the car left the lane also for minor episodes (like the one you show in the video). To avoid reporting too many failed tests we use a tolerance parameter that specifies "how much" of the car must be outside the lane; for example, the car must be at least 50% outside the lane to consider a test to be failed. 

I create #44 to update the documentation and clarify this point.

>  Does the test always continue to the end even if there is a failure at an early stage? Or should the test actually stop when the car goes out of the lane?

No, tests do not continue till the end if an oracle triggers. So, as soon as a test fails, the simulation stops.

**Comment by ezecastellano** [comment]
Since issue [#44](https://github.com/se2p/tool-competition-av/issues/44) was created. I believe this issue should be closed. Thanks for the clear answer.
